waku/v2/discv5/option.go

```diff
@@ -0,0 +1,132 @@
+package discv5
+
+import (
+	"crypto/ecdsa"
+
+	"github.com/ethereum/go-ethereum/p2p/enode"
+	"github.com/multiformats/go-multiaddr"
+	"github.com/prometheus/client_golang/prometheus"
+	cli "github.com/urfave/cli/v2"
+	"github.com/urfave/cli/v2/altsrc"
+	"github.com/waku-org/go-waku/waku/v2/dnsdisc"
+	"go.uber.org/zap"
+)
+
+type DiscV5Parameters struct {
+	// used by localNode
+	AutoUpdate bool
+	UdpPort    uint
+	// for cli
+	clinodes cli.StringSlice
+	//
+	enable        bool
+	autoFindPeers bool
+	bootnodes     []*enode.Node
+	advertiseAddr []multiaddr.Multiaddr
+	loopPredicate func(*enode.Node) bool
+}
+
+func (params *DiscV5Parameters) CliFlags() []cli.Flag {
+	return []cli.Flag{
+		altsrc.NewBoolFlag(&cli.BoolFlag{
+			Name:        "discv5-discovery",
+			Usage:       "Enable discovering nodes via Node Discovery v5",
+			Destination: &params.enable,
+			EnvVars:     []string{"WAKUNODE2_DISCV5_DISCOVERY"},
+		}), altsrc.NewStringSliceFlag(&cli.StringSliceFlag{
+			Name:        "discv5-bootstrap-node",
+			Usage:       "Text-encoded ENR for bootstrap node. Used when connecting to the network. Option may be repeated",
+			Destination: &params.clinodes,
+			EnvVars:     []string{"WAKUNODE2_DISCV5_BOOTSTRAP_NODE"},
+		}), altsrc.NewUintFlag(&cli.UintFlag{
+			Name:        "discv5-udp-port",
+			Value:       9000,
+			Usage:       "Listening UDP port for Node Discovery v5.",
+			Destination: &params.UdpPort,
+			EnvVars:     []string{"WAKUNODE2_DISCV5_UDP_PORT"},
+		}), altsrc.NewBoolFlag(&cli.BoolFlag{
+			Name:        "discv5-enr-auto-update",
+			Usage:       "Discovery can automatically update its ENR with the IP address as seen by other nodes it communicates with.",
+			Destination: &params.AutoUpdate,
+			EnvVars:     []string{"WAKUNODE2_DISCV5_ENR_AUTO_UPDATE"},
+		}),
+	}
+}
```

Comment:
I'm not sure I like this. The cli-flags are a concern for the `waku` command and not for the `waku` package itself (the library should not really be aware of how it is consumed). I'd rather have this as part of a package within cmd/waku